Doc/library/re.rst

```diff
@@ -572,11 +572,8 @@ character ``'$'``.
    Word boundaries are determined by the current locale
    if the :py:const:`~re.LOCALE` flag is used.
 
-   .. note::
-
-      Note that ``\B`` does not match an empty string, which differs from
-      RE implementations in other programming languages such as Perl.
-      This behavior is kept for compatibility reasons.
+   .. versionchanged:: 3.14
+      ``\B`` now matches the whole empty string.
 
```

Comment:
The fix is LGTM.
I suggest explaining in more detail:

``\B`` used to be unable to match empty string. Now it can match, this behavior is consistent with mainstream languages.

Reply:
I know why you use "whole", the `\B` description:

> \B Matches the empty string, but only when it ...

You want to distinguish from "empty string" here.
Let me think about how to describe it more clearly.

Reply:
It is still difficult to avoid contradiction with "\B Matches the empty string, but only when it ...".

Reply:
How about this:

\B used to be unable to match 0-length string `""`/`b""`. Now it can match, this behavior is consistent with mainstream languages.

Reply:
> \B now matches if the input string is empty.

@Alcaro uses the word "input", it looks fine. Candidate:

`\B` now matches empty input string `""`/`b""`, this behavior is consistent with RE implementations in other programming languages such as Perl.

--------------------

The previous test: https://github.com/python/cpython/issues/124130#issuecomment-2484954604
After that, I use ASCII mode for "a", use UNICODE mode for "ю".
The results are same. So this PR is safe.

Tested with: regex module, openjdk, .net, rust, pcre2.
js/golang only support ASCII word (for \w\W\b\B), even in UNICODE mode.

Reply:
I'm sorry, Perl doesn't have (\B+Unicode_mode) bug.
If add a line `use utf8;` at the beginning of Perl script, it works as expected.
ref: https://perldoc.perl.org/utf8